smack-integration-test/src/main/java/org/jivesoftware/smackx/muc/MultiUserChatRolesAffiliationsPrivilegesIntegrationTest.java

```diff
@@ -619,4 +621,295 @@ public void mucPresenceTestForWitnessingKick() throws Exception {
+            XMPPException.XMPPErrorException xe1 = assertThrows(XMPPException.XMPPErrorException.class,
+                            () -> mucAsSeenByTwo.revokeModerator(nicknameOne));
+            // Moderator cannot revoke from Admin
+            XMPPException.XMPPErrorException xe2 = assertThrows(XMPPException.XMPPErrorException.class,
+                            () -> mucAsSeenByThree.revokeModerator(nicknameOne));
+            // Moderator cannot revoke from Owner
+            XMPPException.XMPPErrorException xe3 = assertThrows(XMPPException.XMPPErrorException.class,
+                            () -> mucAsSeenByThree.revokeModerator(nicknameTwo));
+            assertEquals(xe1.getStanzaError().getCondition().toString(), "not-allowed");
+            assertEquals(xe2.getStanzaError().getCondition().toString(), "not-allowed");
+            assertEquals(xe3.getStanzaError().getCondition().toString(), "not-allowed");
+        } finally {
+            tryDestroy(mucAsSeenByOne);
+        }
+    }
+
+    /**
+     * Asserts that an unmoderated room assigns the correct default roles for a given affiliation
+     *
+     * <p>From XEP-0045 § 5.1.2:</p>
+     * <blockquote>
+     * ...the initial default roles that a service SHOULD set based on the user's affiliation...
+     * </blockquote>
+     *
+     * @throws Exception when errors occur
+     */
+    @SmackIntegrationTest
+    public void mucTestDefaultRoleForAffiliationInUnmoderatedRoom() throws Exception {
+        EntityBareJid mucAddress = getRandomRoom("smack-inttest-unmoderatedroles");
+
+        MultiUserChat mucAsSeenByOne = mucManagerOne.getMultiUserChat(mucAddress);
+        MultiUserChat mucAsSeenByTwo = mucManagerTwo.getMultiUserChat(mucAddress);
+        MultiUserChat mucAsSeenByThree = mucManagerThree.getMultiUserChat(mucAddress);
+
+        final Resourcepart nicknameOne = Resourcepart.from("one-" + randomString);
+        final Resourcepart nicknameTwo = Resourcepart.from("two-" + randomString);
+        final Resourcepart nicknameThree = Resourcepart.from("three-" + randomString);
+
+        createMuc(mucAsSeenByOne, nicknameOne.toString());
+        mucAsSeenByTwo.join(nicknameTwo);
+        mucAsSeenByThree.join(nicknameThree);
+
+        final ResultSyncPoint<String, Exception> resultSyncPoint = new ResultSyncPoint<>();
+        mucAsSeenByOne.addParticipantStatusListener(new ParticipantStatusListener() {
+            @Override
+            public void adminGranted(EntityFullJid participant) {
+                resultSyncPoint.signal("done");
+            }
+        });
+        mucAsSeenByOne.grantAdmin(conTwo.getUser().asBareJid());
+        resultSyncPoint.waitForResult(timeout);
+
+        try {
+            assertEquals(mucAsSeenByOne.getOccupantsCount(), 3);
+            assertEquals(MUCRole.moderator, mucAsSeenByOne.getOccupant(
+                            JidCreate.entityFullFrom(mucAddress + "/" + nicknameOne.toString())).getRole());
```

Comment:
Use the `JidCreate.entityFullFrom(EntityBareJid, Resourcepart)` factory method here and everywhere else.

Reply:
Changed in 91897831dd520863fa5adf31565fb73aa836a78e